File: code/go/ch07/src/luago/state/api_vm.go
package state

func (self *luaState) PC() int {
	return self.pc
}

func (self *luaState) AddPC(n int) {
	self.pc += n
}

func (self *luaState) Fetch() uint32 {
	i := self.proto.Code[self.pc]
	self.pc++
	return i
}

func (self *luaState) GetConst(idx int) {
	c := self.proto.Constants[idx]
	self.stack.push(c)
}

func (self *luaState) GetRK(rk int) {
	if rk > 0xFF { // constant
		self.GetConst(rk & 0xFF)
	} else { // register
		self.PushValue(rk + 1)
	}
}

File: code/csharp/ch06/luavm/state/ApiVm.cs
namespace luavm.state
{
    public partial struct LuaState
    {
        public int PC()
        {
            return pc;
        }

        public void AddPC(int n)
        {
            pc += n;
        }

        public uint Fetch()
        {
            var i = proto.Code[pc];
            pc++;
            return i;
        }

        public void GetConst(int idx)
        {
            var c = proto.Constants[idx];
            stack.push(c);
        }

        public void GetRK(int rk)
        {
            if (rk > 0xFF)
            {
                GetConst(rk & 0xFF);
            }
            else
            {
                PushValue(rk + 1);
            }
        }
    }
}